proto/message_contents/public_key.proto

```diff
@@ -26,6 +27,30 @@ message UnsignedPublicKey {
     }
 }
 
+// Signed public key that uses self-describing signatures (aka
+// 'AssociationProof')
+message SignedPublicKeyV2 {
+    bytes key_bytes = 1;  // embeds an UnsignedPublicKey
```

Comment:
Clarifying: Its an explicit choice to have the bytes being associated with a wallet be external to the associationProof?

Reply:
for simplicity I've tried to break down the usages of signatures into different concepts which allows them to be specialized. AssociationProof only addresses a wallet looking to associate a key with its account.

I Imagine signing a message for integrity would be captured in an IntegrityProof which would have different requirements -- Favoring a digest over the input text for example